src/js/common/hooks/usePersistentState.js

```diff
@@ -0,0 +1,34 @@
+import { useCallback, useEffect, useState } from 'react';
+
+/**
+ * Save the value on local storage when it changes and get from local storage when the hook is mounted.
+ *
+ * @param {object} params Params
+ * @param {string} params.key Local storage key
+ * @param {string} params.defaultValue Default value
+ * @param {boolean} params.isJson Is the value a JSON
+ * @returns {Array} [value, setValue, handleClearValue]
+ */
+export const usePersistentState = ({ key, defaultValue, isJson = false }) => {
+  const [value, setValue] = useState(() => {
+    const data = localStorage.getItem(key);
+    if (!data) return defaultValue;
+    if (isJson) return JSON.parse(data) || defaultValue;
+    return data;
+  });
+
+  const handleRestoreDefault = useCallback(() => {
```

Comment:
The functions inside a component are created in every rendering, meaning that in each rendering they are in a different memory address. Generally this is not a problem unless you use one of these functions inside a useEffect, as this will cause an infinite loop (useEffect will run on every render, because the function has changed). That's why in custom hooks I usually use useCallback, because that way the function is memoized and I can use it anywhere without worry. 

Summarizing: In 99% of the time useCallback is useless and cause performance loss, but it's useful when you need to use a function inside an useEffect or creating custom hooks.